integrationTests/multiShard/block/executingMiniblocksSc_test.go

```diff
@@ -688,13 +699,14 @@ func createTxDeploy(
 }
 
 func createTxTopUp(
+	nonce uint64,
 	tn *integrationTests.TestProcessorNode,
 	topUpVal *big.Int,
 	scAddress []byte,
 ) *transaction.Transaction {
 
 	tx := &transaction.Transaction{
-		Nonce:    0,
+		Nonce:    nonce,
 		Value:    topUpVal,
 		RcvAddr:  scAddress,
 		SndAddr:  tn.OwnAccount.PkTxSignBytes,
```

Comment:
OwnAccount has Balance as field and it can get updated separately to see if the value in the blockchain is the same. This is done for the singleShard SC test and for the long tests. The same could be done with the Nonce. Add the Nonce field to the TestWallettAccount structure, and increase its value there. Then you can eliminate the nonce uint64 as a parameter from a lot of functions.

Reply:
Good catch

Reply:
Done